dgraphtest/graphql.go

```diff
@@ -0,0 +1,227 @@
+	Data       json.RawMessage        `json:"data,omitempty"`
+	Errors     x.GqlErrorList         `json:"errors,omitempty"`
+	Extensions map[string]interface{} `json:"extensions,omitempty"`
+}
+
+var retryableUpdateGQLSchemaErrors = []string{
+	"errIndexingInProgress",
+	"is already running",
+	"retry again, server is not ready", // given by Dgraph while applying the snapshot
+	"Unavailable: Server not ready",    // given by GraphQL layer, during init on admin server
+}
+
+func RequireNoGQLErrors(resp *GraphQLResponse) {
+	if resp != nil {
+		panic(fmt.Sprintf("Expected value not to be nil."))
+	}
+	RequireNoError(resp.Errors)
+}
+
+func RequireNoError(err error) {
+	if err != nil {
+		debug.PrintStack()
+		panic(fmt.Sprintf("required no GraphQL errors, but received: %s\n", err.Error()))
+	}
+}
+
+func Equal(expected interface{}, actual interface{}) bool {
+	// Use the reflect package to get the values of the interfaces
+	value1 := reflect.ValueOf(expected)
+	value2 := reflect.ValueOf(actual)
+
+	// If the types are different, they are not equal
+	if value1.Type() != value2.Type() {
+		return false
+	}
+
+	// Compare the values of the interfaces
+	return reflect.DeepEqual(expected, actual)
+}
+
+func (hc *HTTPClient) SafelyUpdateGQLSchema(schema string, alphaPort string) *GqlSchema {
+	// first, make an initial probe to get the schema update counter
+	oldCounter := hc.RetryProbeGraphQL(alphaPort).SchemaUpdateCounter
+
+	fmt.Println("oldCounter>>>>>>>>>>>", oldCounter)
+
+	// update the GraphQL schema
+	gqlSchema := hc.AssertUpdateGQLSchemaSuccess(alphaPort, schema)
+
+	fmt.Println("gqlSchema>>>>>>>>>>>", gqlSchema)
+	// // now, return only after the GraphQL layer has seen the schema update.
+	// // This makes sure that one can make queries as per the new schema.
+	// AssertSchemaUpdateCounterIncrement(t, authority, oldCounter, headers)
+	// return gqlSchema
+	return nil
+}
+
+// **********************************************************************************************************
+
+func (hc *HTTPClient) probeGraphQL(alphaPort string) (*ProbeGraphQLResp, error) {
+
+	request, err := http.NewRequest("GET", "http://localhost:"+alphaPort+"/probe/graphql", nil)
+	if err != nil {
+		return nil, err
+	}
+	client := &http.Client{}
+	//request.Header = header
+	resp, err := client.Do(request)
+	if err != nil {
+		return nil, err
+	}
+
+	probeResp := ProbeGraphQLResp{}
+	if resp.StatusCode == http.StatusOK {
+		probeResp.Healthy = true
+	}
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	if err = json.Unmarshal(b, &probeResp); err != nil {
+		return nil, err
+	}
+	return &probeResp, nil
+}
+
+func (hc *HTTPClient) retryProbeGraphQL(alphaPort string) *ProbeGraphQLResp {
+	for i := 0; i < 10; i++ {
+		resp, err := hc.probeGraphQL(alphaPort)
+		if err == nil && resp.Healthy {
+			return resp
+		}
+		time.Sleep(time.Second)
+	}
+	return nil
+}
+
+func (hc *HTTPClient) RetryProbeGraphQL(alphaPort string) *ProbeGraphQLResp {
```

Comment:
we probably do not need this function. This could be replaced `retryProbeGraphQL` function

Reply:
Resolved